### ISSUE TITLE
Introduce `legalName` in the API

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#4581]
Part of https://github.com/opencollective/opencollective/issues/3689

Items with the :test_tube: icon are things that should be covered by unit tests.

- [x] Add a `legalName` (string, default=null) field on the `Collectives` model
- [x] The mutation(s) to edit accounts should accept a `legalName` as argument :test_tube:
- [x] On GQLV2 `Account` object, add a `legalName` field. The resolver should take care of checking the permissions :test_tube::
  - [x] Field should always be visible for admin(s) of the account
  - [x] Field should always be visible for host accounts
  - [x] We should create a new `SEE_LEGAL_NAME` context permission and check it with `getContextPermission`
- [x] Use `setContextPermission` for `SEE_LEGAL_NAME` in all contexts where we want to allow it :test_tube:
  - [x] For expense payees, if `remoteUser` has the host admin permissions
  - [x] For financial contributors, if `remoteUser` has the host admin permissions
- [x] When generating the tax form on HelloWorks, use the `legalName` if available :test_tube:
- [x] When submitting an expense, if `legalName` is defined, make sure it matches the payout method (bank account) :test_tube:

**Comment by Betree** [comment]
Extracted CSV implementation to a [separate issue](https://github.com/opencollective/opencollective/issues/4710) since it's a different service. This one is all done!

**Comment by Betree** [comment]
Oops, we still need to merge https://github.com/opencollective/opencollective-api/pull/6545 for it to be considered completed.

**Comment by SudharakaP** [comment]
> Oops, we still need to merge [opencollective/opencollective-api#6545](https://github.com/opencollective/opencollective-api/pull/6545) for it to be considered completed.

Yes, I missed this one; I'll go through your [final review](https://github.com/opencollective/opencollective-api/pull/6545#pullrequestreview-757131842) for this today and hopefully we can merge it this week. 🙂

**Comment by Betree** [comment]
This is done
